### PR TITLE
The number of netty IO threads is set according to the number of available cores

### PR DESCRIPTION
**Pull request by mcdull-zhang**
### What changes were proposed in this pull request?

The number of netty IO threads is set according to the number of available cores.

For example in the case of executorCores=3:

**before**
image

**after**
image2022-10-21 16_39_31

### Why are the changes needed?

Avoid wasting resources

### How was this patch tested?

/cc @waitinfuture

### CODE REVIEW

**Comment by waitinfuture** [comment]
please run ./dev/reformat to automatically reformat style

**Comment by waitinfuture** [comment]
I think we can add a config to CelebornConf in the constructor of RssShuffleManager and use the conf in ShuffleClient.get(), so we can avoid adding the numCores through the call-chain. Like the following
```
  public RssShuffleManager(SparkConf conf) {
    this.conf = conf;
    this.celebornConf = SparkUtils.fromSparkConf(conf);
    this.cores = conf.getInt(SparkLauncher.EXECUTOR_CORES, 1);
    celebornConf.set("celeborn.dispatcher.threads", numCores);
    this.fallbackPolicyRunner = new RssShuffleFallbackPolicyRunner(celebornConf);
  }
```

cc @pan3793

**Comment by waitinfuture** [comment]
> 

We should first define a config in CelebornConf like the following before use the key
```
  val RPC_DISPATCHER_THREADS: ConfigEntry[Int] =
    buildConf("celeborn.client.rpc.dispatcher.threads")
      .categories("client")
      .version("0.2.0")
      .doc("xxxxx.")
      .intConf
      .createWithDefault(16)
```

**Comment by pan3793** [comment]
It's unintuitive that the user-specific value has lower priority than configured one `rss.rpc.io.threads`

https://github.com/alibaba/celeborn/blob/399236c880a79d5e6dcf7d591fbc6653bcfe8150/common/src/main/scala/org/apache/celeborn/common/rpc/netty/NettyRpcEnv.scala#L53-L56

**Comment by pan3793** [comment]
> > 
> 
> We should first define a config in CelebornConf like the following before use the key
> 
> ```
>   val CLIENT_RPC_DISPATCHER_THREADS: ConfigEntry[Int] =
>     buildConf("celeborn.client.rpc.dispatcher.threads")
>       .categories("client")
>       .version("0.2.0")
>       .doc("xxxxx.")
>       .intConf
>       .createWithDefault(16)
> ```

+1 to migrate conf to `CelebornConf`, for naming, I would suggest to respect Spark style, just simply change the `rss.` to `celeborn.`.

File: client/src/test/java/org/apache/celeborn/client/ShuffleClientSuiteJ.java [hunk 1]
```diff
@@ -195,7 +195,7 @@ private CelebornConf setupEnv(CompressionCodec codec) throws IOException, Interr
     conf.set("rss.client.compression.codec", codec.name());
```
Comment (pan3793) [inline client/src/test/java/org/apache/celeborn/client/ShuffleClientSuiteJ.java:195]: It should be changed to "celeborn.shuffle.compression.codec", would you please change it to `SHUFFLE_COMPRESSION_CODEC.key`?

File: worker/src/test/scala/org/apache/celeborn/service/deploy/cluster/ReadWriteTestBase.scala [hunk 2]
```diff
@@ -41,8 +41,8 @@ trait ReadWriteTestBase extends Logging {
       .set("rss.client.compression.codec", codec.name)
```
Comment (pan3793) [inline worker/src/test/scala/org/apache/celeborn/service/deploy/cluster/ReadWriteTestBase.scala:41]: ditto

File: client-spark/spark-3/src/main/java/org/apache/spark/shuffle/celeborn/RssShuffleManager.java [hunk 3]
```diff
@@ -53,7 +54,8 @@ public class RssShuffleManager implements ShuffleManager {
   public RssShuffleManager(SparkConf conf) {
     this.conf = conf;
     this.celebornConf = SparkUtils.fromSparkConf(conf);
-    this.cores = conf.getInt(SparkLauncher.EXECUTOR_CORES, 1);
+    this.executorCores = conf.getInt(SparkLauncher.EXECUTOR_CORES, 1);
+    this.driverCores = conf.getInt("spark.driver.cores", 0);
```
Comment (pan3793) [inline client-spark/spark-3/src/main/java/org/apache/spark/shuffle/celeborn/RssShuffleManager.java:58]: I suggest renaming `conf: SparkConf` to `sparkConf: SparkConf`, and renaming `celebornConf: CelebornConf` to `conf: CelebornConf`, to be consistent w/ other places.

**Comment by waitinfuture** [comment]
> 

In spark, dispatcher's thread number is different from transportConf.numUsableCores. The former is just numUsableCores, while the latter is rss.rpc.io.threads (and fallback to numUsableCores). So I suggest to use different configs for them. We should remain celeborn.rpc.io.threads for transportConf.numUsableCores, and use celeborn.rpc.dispatcher.threads for dispatcher threads.

**Pull request closed without merging**